sdk/main/src/ED25519PublicKey.cc

```diff
@@ -112,7 +112,8 @@ bool ED25519PublicKey::verifySignature(const std::vector<unsigned char>& signatu
 
   if (verificationResult <= 0)
   {
-    std::cout << "Failed to verify signature with code [" << verificationResult << ']' << std::endl;
+    std::string message = "Failed to verify signature with code [" + std::to_string(verificationResult) + ']';
+    throw std::runtime_error(message.c_str());
```

Comment:
You need to free the `messageDigestContext` here before throwing. Something like `EVP_MD_CTX_free(messageDigestContext)`.

Reply:
Thanks for noticing it. I fixed it in 04f7036023e74d9f62f24cd49f49801eabab2d2e.